src/Microsoft.Azure.WebJobs.Host/Bindings/SystemBindingData.cs

```diff
@@ -0,0 +1,84 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using Microsoft.Azure.WebJobs.Host.Bindings.Path;
+
+namespace Microsoft.Azure.WebJobs.Host.Bindings
+{
+    /// <summary>
+    /// Built in object for all binding expressions. 
+    /// </summary>
+    /// <remarks>
+    /// It's expected this class is created and added to the binding data. 
+    /// </remarks>
+    internal class SystemBindingData
+    {
+        // The public name for this binding in the binding expressions. 
+        public const string Name = "sys";
+
+        // An internal name for this binding that uses characters that gaurantee it can't be overwritten by a user. 
+        // This ensures that we can always unambiguously retrieve this later. 
+        private const string InternalKeyName = "$sys";
```

Comment:
Not sure why you don't want to use the MS_ prefix for disambiguation, which is a precident we're already using in several places. E.g. bindings like http (request properties), in our trace event stuff (ScopeKeys), etc.